src/SldStyleParser.ts

```diff
@@ -292,188 +284,203 @@ export class SldStyleParser implements StyleParser<string> {
-      default:
-        throw new Error('Unsupported expression.');
-    }
-  }
-
-  getLiteralFromSldObject(sldObject: any):
-  LiteralValue<string> | LiteralValue<number> | LiteralValue<boolean> | LiteralValue<null> {
-    const result = {
-      type: 'literal',
-      value: sldObject._
-    };
-    const num = parseFloat(sldObject._);
-    if (!Number.isNaN(num)) {
-      result.value = num;
-    }
-    if (result.value === 'true') {
-      result.value = true;
-    }
-    if (result.value === 'false') {
-      result.value = false;
-    }
-    return result as LiteralValue<string> | LiteralValue<number> | LiteralValue<boolean> | LiteralValue<null>;
-  }
-
-  getFunctionFromSldObject(sldObject: any): FunctionCall {
-    return {
-      type: 'functioncall',
-      name: sldObject.$.name,
-      args: sldObject.$$.map((sldArg: any) => this.getExpressionFromSldObject(sldArg))
-    };
-  }
-
-  getPropertyNameFromSldObject(sldObject: any): PropertyName {
+  sldObjectToGeoStylerStyle(sldObject: any): Style {
+    const rules = this.getRulesFromSldObject(sldObject);
+    const name = this.getStyleNameFromSldObject(sldObject);
     return {
-      type: 'property',
-      name: sldObject._
+      name,
+      rules
     };
   }
 
-  getValueFromSldObject(sldObject: any): string {
-    return sldObject._;
-  }
-
-  getExpressionOrValueFromSldObject(sldObject: any): Expression | string | undefined {
-    if (!sldObject) {
-      return undefined;
-    }
-    const expressionOrValue =_get(sldObject, '$$[0]');
-    const expressionOrValueName = _get(sldObject, '$$[0]["#name"]');
-    if (expressionOrValueName === '__text__') {
-      return this.getValueFromSldObject(expressionOrValue);
-    } else {
-      return this.getExpressionFromSldObject(expressionOrValue);
-    }
+  /**
+   * Get the geostyler-style rules from a SLD Object (created with fast-xml-parser).
+   *
+   * @param sldObject The SLD object representation (created with fast-xml-parser)
+   * @return The geostyler-style rules
+   */
+  getRulesFromSldObject(sldObject: any): GsRule[] {
+    const layers = getChildren(sldObject[0].StyledLayerDescriptor, 'NamedLayer');
+    const rules: GsRule[] = [];
+    layers.forEach(({NamedLayer: layer}: any) => {
+      getChildren(layer, 'UserStyle').forEach(({UserStyle: userStyle}: any) => {
+        getChildren(userStyle, 'FeatureTypeStyle').forEach(({FeatureTypeStyle: featureTypeStyle}: any) => {
+          getChildren(featureTypeStyle, 'Rule').forEach(({Rule: sldRule}: any) => {
+            const filter: GsFilter | undefined = this.getFilterFromRule(sldRule);
+            const scaleDenominator: ScaleDenominator | undefined = this.getScaleDenominatorFromRule(sldRule);
+            const symbolizers: Symbolizer[] = this.getSymbolizersFromRule(sldRule);
+            const ruleTitle = get(sldRule, 'Title.#text');
+            const ruleName = get(sldRule, 'Name.#text');
+            const name = ruleTitle !== undefined
```

Comment:
According to the SLD specification, Rules do not have a `title` property, but only an optional `name` property. So I guess this logic here is obsolete. Might have confused this with the style name and style title?